jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/ImageName.java

```diff
@@ -271,6 +288,7 @@ public static void validate(String image) {
     // Validate parts and throw an IllegalArgumentException if a part is not valid
     private void doValidate() {
         List<String> errors = new ArrayList<>();
+        Pattern digestReg = Pattern.compile("^sha" + hashLen + ":[a-z0-9]{" + hashLen/8 + ",}$");
```

Comment:
instead of creating a digestReg variable why not modify DIGEST_REGEXP ?

Probably we can create two regexp for this:
- DIGEST_256_REGEXP
- DIGEST_512_REGEXP

then DIGEST_REGEXP can be either of these two algorithms:
private static final String DIGEST_REGEXP = "^(?:" + DIGEST_256_REGEXP + "|" + DIGEST_512_REGEXP + ")";

Reply:
I don't know I'm right with approach or not but If we have strictly these two algorithms(As of Image name specifications) then this is the best approach to do like this:
 private static final String DIGEST_REGEXP = "^(?:" + DIGEST_256_REGEXP + "|" + DIGEST_512_REGEXP + ")";
But I was thinking in the way that if in future Image Container specifications will extend with sha then may we have to add one more regex here, yes can update this as  per requirements.
For now I'll modify this according to SHA256 and SHA512 only.

Reply:
Done !